invenio_rdm_records/resources/serializers.py

```diff
@@ -23,6 +23,16 @@ def _serialize_access_right(self, obj):
                 category=obj["access"]["access_right"]
             ))
 
+    def _serialize_localized_publication_date(self, obj):
+        """Dump ui config for `access_right` field."""
+        localized_pub_date = dict(
+            publication_date_l10n=dict(
+                value=obj["metadata"]["publication_date_l10n"]
+            ))
+        del obj["metadata"]["publication_date_l10n"]
```

Comment:
@lnielsen this is the point I referred to about not so clean way of handling the serialization. In principle, the new added field is serializing the `publication_date_l10n` already, so this might be redundant here...